python/ray/serve/tests/test_healthcheck.py

```diff
@@ -202,6 +203,27 @@ def check_fails_3_times():
     check_fails_3_times()
 
 
+def test_health_check_failure_makes_deployment_unhealthy(serve_instance):
```

Comment:
add a test:
The deployment is `HEALTHY` -> `UNHEALTHY` when there is a replica failing the health check.

Reply:
Done!